### PR TITLE
add more tests for SAN conversion and parsing

### PR DESCRIPTION
**Pull request by robertnurnberg**
Master will fail some of these new tests when running `meson test -C build --test-args='--test-suite="SAN Parser"'
`, so this PR should probably only be merged once they are fixed.

I also corrected a misnamed Bishop test (was duplicate of a King move), and added SAN parsing tests where they were missing.

If you really wanted to parse e.g. `Nd4b3` instead of the correct `N4b3`, then we could add a separate test with a comment to this effect.

Also added some new lines to stick to prevailing formatting of the file.

### CODE REVIEW

File: tests/san.cpp [hunk 1]
```diff
@@ -298,22 +326,142 @@ TEST_SUITE("SAN Parser") {
...
+        }
+    }
+
+    TEST_CASE("Test Rook Ambiguity 2") {
+        auto b = Board{"1r5k/5Rp1/5p1p/1r2Nb1P/2B5/2P1R1P1/3r1PK1/8 b - - 0 1"};
+
+        {
+            auto m = Move::make(Square::underlying::SQ_B5, Square::underlying::SQ_B2);
+
+            CHECK(uci::moveToSan(b, m) == "Rb2");
```
Comment (Disservin) [inline tests/san.cpp:420]: this one is incorrect, the actual string should be "Rbb2" since both rooks on d and b can move there, I think you dropped one b while making this?
https://lichess.org/analysis/fromPosition/1r5k/5Rp1/5p1p/1r2Nb1P/2B5/2P1R1P1/3r1PK1/8_b_-_-_0_1#1

**Comment by Disservin** [comment]
I added a fix for the ambiguous move detection and also changed the one wrong test

File: tests/san.cpp (hunk 1 above)
Comment (robertnurnberg) [reply tests/san.cpp]: I forgot the remove the black rook on d2 for this test, yeah. will fix the FEN later.

**Comment by robertnurnberg** [comment]
Nice, that was quick. I added as a further test the `Rb2` test I had in mind.
